js_modules/dagster-ui/packages/ui-core/src/insights/ActivityChart.tsx

```diff
@@ -0,0 +1,196 @@
+            const unitLower = unit.toLowerCase();
+            if (unitLower === 'seconds' || unitLower === 'milliseconds') {
+              return formatDuration(value, {unit: unitLower})[0];
+            }
+            return {value, unit};
+          })();
+
+          return (
+            <Popover
+              key={index}
+              targetTagName="div"
+              interactionKind="hover"
+              popoverClassName={styles.popover}
+              placement="top"
+              content={
+                <TooltipCard>
+                  <Box flex={{direction: 'column'}}>
+                    <Box border="bottom" padding={{horizontal: 12, vertical: 8}}>
+                      <Subheading>
+                        {formatDate(new Date(date + index * 60 * 60 * 1000), {
+                          month: 'short',
+                          day: 'numeric',
+                          hour: 'numeric',
+                          minute: 'numeric',
+                        })}
+                      </Subheading>
+                    </Box>
+                    <Box
+                      flex={{direction: 'row', alignItems: 'center', gap: 4}}
+                      padding={{horizontal: 12, vertical: 8}}
+                    >
+                      <Mono>{numberFormatterWithMaxFractionDigits(2).format(displayValue)}</Mono>
+                      <Body color={Colors.textLight()}>{displayUnit}</Body>
+                    </Box>
+                    {value > 0 && onClick ? (
+                      <Box padding={{horizontal: 12, vertical: 8}} border="top">
+                        <Body color={Colors.textLight()}>Click to view details</Body>
+                      </Box>
+                    ) : null}
+                  </Box>
+                </TooltipCard>
+              }
+            >
+              <div
+                className={styles.tileContainer}
+                style={
+                  {
+                    '--tile-hover-color': hoverColor,
+                    '--tile-color': color,
+                  } as React.CSSProperties
+                }
+              >
+                <div className={styles.placeholderTile} />
+                {opacity ? (
+                  <button
+                    className={styles.tileButton}
+                    style={{opacity}}
+                    onClick={() => {
+                      if (onClick) {
+                        onClick({
+                          before: date / 1000 + index * 60 * 60,
+                          after: date / 1000 + (index - 1) * 60 * 60,
+                          metric,
+                          unit,
+                        });
+                      }
+                    }}
+                  />
+                ) : null}
+              </div>
+            </Popover>
+          );
+        })}
+      </div>
+    </div>
+  );
+};
+
+export const ActivityChartRow = memo(InnerActivityChartRow) as typeof InnerActivityChartRow;
```

Comment:
This is a generic ActivityChart component, to be used by both assets and jobs. It is based very heavily on `AssetCatalogActivityChart`, deleted above.